assets/js/components/Dashboard/NewPaMessage/NewPaMessagePage.tsx

```diff
@@ -314,4 +389,88 @@ const ReviewAudioButton = ({
   );
 };
 
+interface NewPaMessageHeaderProps {
+  associatedAlert: AlertModel;
+  visualText: string;
+  phoneticText: string;
+  navigateTo: (page: Page) => void;
+  setImportMessage: Dispatch<SetStateAction<boolean>>;
+  setImportLocations: Dispatch<SetStateAction<boolean>>;
+  setEndWithEffectPeriod: Dispatch<SetStateAction<boolean>>;
+  setAssociatedAlert: Dispatch<SetStateAction<AlertModel>>;
+  setVisualText: Dispatch<SetStateAction<string>>;
+  setPhoneticText: Dispatch<SetStateAction<string>>;
+  setEndDate: Dispatch<SetStateAction<string>>;
+  setEndTime: Dispatch<SetStateAction<string>>;
+}
+
+const NewPaMessageHeader = ({
+  associatedAlert,
+  navigateTo,
+  setImportMessage,
+  setImportLocations,
+  setEndWithEffectPeriod,
+  setAssociatedAlert,
+  setVisualText,
+  setPhoneticText,
+  setEndDate,
+  setEndTime,
+}: NewPaMessageHeaderProps) => {
+  const now = moment();
+  const formatActivePeriod = (activePeriods: ActivePeriod[]) => {
+    const [start, end] = getAlertEarliestStartLatestEnd(activePeriods);
+    return (
+      <div className="effect-period">
+        Alert Effect period: {start} - {end}
+      </div>
+    );
+  };
+
+  return associatedAlert.id ? (
+    <Row md="auto" className="align-items-center">
+      <div className="associated-alert-header">
+        Associated Alert: Alert ID {associatedAlert.id}
+        <Button variant="link" onClick={() => navigateTo(Page.ALERTS)}>
+          Edit
+        </Button>
+        <Button
+          variant="link"
+          onClick={() => {
```

Comment:
With sub-components like this, I find it's helpful to think in terms of passing callbacks (as opposed to setters), which sometimes leads to some natural abstraction. In this case, if we passed a single `onClear` callback, we could perform all these state updates in the parent's lexical scope, which would eliminate a lot of boilerplate (passing, destructuring, type signatures)